### PR TITLE
feat(quantic): recommendations controller APEX class created

### PR DESCRIPTION
**Pull request by mmitiche**
## [SFINT-5401](https://coveord.atlassian.net/browse/SFINT-5401)

- To simplify the implementation of the recommendations feature in Quantic I created the new RecommendationsController APEX Class that will be called by the `QuanticRecommendationsInterface` in order to get the Headless configuration that will power the Headless Recommendations engine.

[SFINT-5401]: https://coveord.atlassian.net/browse/SFINT-5401?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SimonMilord — COMMENTED** [review]
Not too sure I understand the reasoning behind this PR. Seems like we are mostly duplicating the code from the HeadlessController class. 

Even after reading the JIRA description I am not sure what this is adding that is simpler

**Comment by SimonMilord** [comment]
> > Not too sure I understand the reasoning behind this PR. Seems like we are mostly duplicating the code from the HeadlessController class.
> > Even after reading the JIRA description I am not sure what this is adding that is simpler
> 
> When implementing the Quantic Recommendations feature, the user needs to generate his own search token to perform the queries, he will also need to enforce a search hub in this token, exactly like he needs to do this when implementing the search feature.
> 
> However, the user will most likely use two separate search tokens for the search use case and the recommendations use case since it's likely that he we will need a specific search hub value for the search use case and a different search hub value for the recommendation use case.
> 
> I order to do this the QuanticRecommendationsInterface component needs to grab the Headless configuration, that includes the token, from a class different from the class used to grab the headless configuration in the search use case otherwise it wouldn't be possible to generate two separate search tokens.
> 
> So in this PR I'm making the QuanticRecommendationsInterface rely on the RecommendationController so when the user wants to implement this feature he would just need to modify this latter class to generate the token he wants in the recommendation use case. exactly like he would modify the HeadlessController class to generate the token he wants in the search use case.

Gotcha, thank you for clarifying, makes more sense now!

File: packages/quantic/force-app/main/default/classes/RecommendationsController.cls [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Simple Apex Controller to fetch data for Headless initialization specific to the Recommendations use case.
```
Comment (SimonMilord) [inline packages/quantic/force-app/main/default/classes/RecommendationsController.cls:2]:
```suggestion
 * Simple Apex Controller to fetch the data for Headless initialization specific to the Recommendations use case.
```

**Review by SimonMilord — APPROVED** [review]
lgtm!

File: packages/quantic/quantic-examples-community/profiles/Quantic Examples Profile.profile-meta.xml
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/quantic-examples-community/profiles/Quantic Examples Profile.profile-meta.xml:1]: oh wow good thinking!

[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/quantic-examples-community/profiles/Quantic Examples Profile.profile-meta.xml:1]: the .profile file was deleted and replaced with .profile-meta.xml?

File: packages/quantic/force-app/main/default/classes/RecommendationsControllerTest.cls [hunk 2]
```diff
@@ -0,0 +1,13 @@
+@isTest
+private class RecommendationsControllerTest {
+  static final String sampleHeadlessConfiguration = '{"platformUrl":"https://platform.cloud.coveo.com","accessToken":"xx564559b1-0045-48e1-953c-3addd1ee4457","organizationId":"searchuisamples"}';
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/classes/RecommendationsControllerTest.cls:3]:
```suggestion
  // This is a demo token and it's okay to be public.
  static final String sampleHeadlessConfiguration = '{"platformUrl":"https://platform.cloud.coveo.com","accessToken":"xx564559b1-0045-48e1-953c-3addd1ee4457","organizationId":"searchuisamples"}';
```

**Review by erocheleau — APPROVED** [review]
I had the same question as Simon but this makes a lot of sense :)

File: packages/quantic/quantic-examples-community/profiles/Quantic Examples Profile.profile-meta.xml
[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/quantic/quantic-examples-community/profiles/Quantic Examples Profile.profile-meta.xml:1]: exactly! apparently that's the new format Salesforce use.

**Pull request merged**